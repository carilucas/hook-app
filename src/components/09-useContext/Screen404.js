import React from 'react'

export const Screen404 = () => {
   return (
      <div>
         <h1 className="text-center mt-5">Error 404</h1>
      </div>
   )
}
