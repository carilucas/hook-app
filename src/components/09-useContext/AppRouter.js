import React from 'react';
import {
BrowserRouter as Router,
   Switch,
   Route,
   Redirect,
} from "react-router-dom";
import { NavBar } from "./NavBar";
import { AboutScreen } from './AboutScreen';
import { HomeScreen } from './HomeScreen';
import { LoginScreen } from './LoginScreen';
import { Screen404 } from './Screen404';

export const AppRouter = () => {
   return (
      <Router>
         <div>
            <NavBar />
            <main className="container">
               <Switch>
                  <Route exact path="/" component={ HomeScreen }/>
                  <Route exact path="/about" component= { AboutScreen }/>
                  <Route exact path="/login" component={ LoginScreen }/>
                  <Route path="/not-found" component={ Screen404 }/>
                  <Redirect to="./not-found" />
               </Switch>
            </main>
         </div>
      </Router>
   )
}
