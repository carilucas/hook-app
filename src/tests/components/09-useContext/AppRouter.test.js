import { mount } from 'enzyme';
import React from 'react';
import { AppRouter } from '../../../components/09-useContext/AppRouter';
import { UserContext } from '../../../components/09-useContext/UserContext';


describe('Pruebas en <AppRouter/>', () => {
   const user = {
      name:'carlos',
      email:'c@ca.com'
   }
   const wrapper = mount(
      <UserContext.Provider value={{user}}>
         <AppRouter/>
      </UserContext.Provider>
   );

   test('Mostrar el snapshot', () => {
     expect(wrapper).toMatchSnapshot(); 
   })
   
})
