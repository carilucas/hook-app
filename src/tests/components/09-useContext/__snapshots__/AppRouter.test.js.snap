// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AppRouter/> Mostrar el snapshot 1`] = `
<div>
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark mb-5"
  >
    <div
      className="container-fluid"
    >
      <a
        aria-current="page"
        className="navbar-brand active"
        href="/"
        onClick={[Function]}
        style={Object {}}
      >
        Navbar
      </a>
      <div
        className="collapse navbar-collapse justify-content-end"
        id="navbarNavAltMarkup"
      >
        <div
          className="navbar-nav"
        >
          <a
            aria-current="page"
            className="nav-link active"
            href="/"
            onClick={[Function]}
            style={Object {}}
          >
            Home
          </a>
          <a
            aria-current={null}
            className="nav-link"
            href="/about"
            onClick={[Function]}
          >
            About
          </a>
          <a
            aria-current={null}
            className="nav-link"
            href="/login"
            onClick={[Function]}
          >
            Login
          </a>
        </div>
      </div>
    </div>
  </nav>
  <main
    className="container"
  >
    <div>
      <h1>
        HomeScreen
      </h1>
      <hr />
      <pre>
        {
   "name": "carlos",
   "email": "c@ca.com"
}
      </pre>
    </div>
  </main>
</div>
`;
